libvast/vast/system/generator_command.hpp

```diff
@@ -0,0 +1,92 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <string_view>
+
+#include <caf/scoped_actor.hpp>
+#include <caf/typed_actor.hpp>
+#include <caf/typed_event_based_actor.hpp>
+
+#include "vast/concept/parseable/to.hpp"
+#include "vast/concept/parseable/vast/expression.hpp"
+#include "vast/concept/parseable/vast/schema.hpp"
+#include "vast/defaults.hpp"
+#include "vast/detail/make_io_stream.hpp"
+#include "vast/expression.hpp"
+#include "vast/logger.hpp"
+#include "vast/system/reader_command_base.hpp"
+#include "vast/system/reader_command_base.hpp"
+#include "vast/system/signal_monitor.hpp"
+#include "vast/system/source.hpp"
+#include "vast/system/tracker.hpp"
+
+namespace vast::system {
+
+/// Default implementation for import sub-commands. Compatible with Bro and MRT
+/// formats.
+/// @relates application
+template <class Generator>
+class generator_command : public reader_command_base {
+public:
+  generator_command(command* parent, std::string_view name)
+      : reader_command_base(parent, name) {
+    add_opt<std::string>("schema,s", "path to alternate schema");
+    add_opt<size_t>("seed", "the random seed");
+    add_opt<size_t>("num,N", "events to generate");
+  }
+
+protected:
+  expected<caf::actor> make_source(caf::scoped_actor& self,
+                                   const caf::config_value_map& options,
+                                   argument_iterator begin,
+                                   argument_iterator end) override {
+    VAST_TRACE(VAST_ARG("args", begin, end));
+    auto seed = get_or(options, "seed", 0u);
```

Comment:
Isn't this going to produce always the same events if a user doesn't provide a seed? I'd use `get_if` here instead and get a random seed from `std::random_device` in case no seed gets provided.

Reply:
Seems reasonable as long as manually supplying a seed makes the runs reproducible.